etc/v4l2loopback_setup.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+if [ "$(whoami)" == "root" ]; then
+	echo "Please do not run this script directly as root. Use your normal development user account."
+	exit 1
+fi
+
+if ! apt-get --version >/dev/null 2>&1; then
+  echo "Unable to find APT package handling utility (apt-get)"
+  echo "Are you sure you're on a Debian-based Linux variant (e.g. Ubuntu)?"
+  exit 1
+fi
+
+install_dependencies() {
+  sudo bash -c "apt-get install --assume-yes kmod git make gcc libelf-dev"
+}
+
+install_kernel_headers() {
+  echo "updating kernel module..."
+  sudo bash <<-EOS
+	sudo apt-get --assume-yes update
+	sudo apt-get --assume-yes dist-upgrade
+	sudo apt-get --assume-yes upgrade
```

Comment:
Is `module-assistant` the thing to be using here? https://github.com/umlaeute/v4l2loopback#distributions

Reply:
You _could_ but it's _a lot_ more involved. I spent the better part of yesterday trying to work it out. The recommended way to install kernel headers is using ` sudo apt install raspberrypi-kernel-headers` which will install the latest kernel headers, _not_ the version for the one currently running ([source](https://www.raspberrypi.com/documentation/computers/linux_kernel.html#kernel-headers)).

I agree, though, and will try to work on installing the headers for the current kernel version today. That would avoid the `(dist-)upgrade` and reboot. Yesterday I was working with a library ([rpi-source](https://github.com/RPi-Distro/rpi-source)) that does a lot of this for you but expects an absurd amount of user input along the way. Will get back to it.

Reply:
> Is `module-assistant` the thing to be using here? https://github.com/umlaeute/v4l2loopback#distributions

Unfortunately that doesn't work that well. The Raspberry Pi installs kernel version ~5.15.0 if you follow the instructions in app. Running `module-assistant` looks for your kernel version and finds the closest kernel version it can download which for me always resolves to ~v5.10.0. This seems to be a common problem from search online too. Your kernel version and the kernel headers must have the _exact_ same version down to the fix version and both of the automated ways install an additional module and download the source headers for that. `modprobe` will then use `uname -r` to find the kernel version running and look for headers with that version and complain that they don't exists. It's frustratingly tricky to get this right.

Reply:
OK, thanks for looking into it!